graph2tac/loader/data_classes.py

```diff
@@ -1,118 +1,79 @@
-from numpy.typing import NDArray
 import numpy as np
 from dataclasses import dataclass
-
-def custom_dataclass_repr(obj) -> str:
-    """String repr for a dataclass, but removing the array data."""
-    kws = {k: f"array(shape={v.shape}, dtype={v.dtype})" if isinstance(v, np.ndarray) else repr(v) for k,v in obj.__dict__.items()}
-    return obj.__class__.__qualname__ + "(" + ", ".join(k + "=" + v for k,v in kws.items()) + ")"
-
-
-@dataclass(repr=False)
-class LoaderGraph:
-    """A single graph as it comes from the loader"""
-    nodes: NDArray[np.uint32]  # [nodes]
-    """Node class labels.  Shape: [number of nodes]"""
```

Comment:
Maybe we can keep the docs, even if they are not official docstrings?

Reply:
Also, why can't we just do what Lasse had originally suggested with the named tuples.  Then we can have both namedtuples and good editor hints including doc strings.  Is there something that makes that approach not work?

Reply:
```
from typing import NamedTuple

class Card(NamedTuple):
    """This is a card type."""
    suit: str
    rank: str
```

Reply:
Or is it that this has too much boilerplate since it requires writing code for two classes instead of one?

Reply:
I had some trouble with it with my first experiments but I admit it probably would work. On the other hand, we need the tensorflow typing anyway, and I prioritize having nice way of managing tensorflow typings over editor features. This is mainly a helper datastructure between pytact, and the GraphTensor, and when we decide to put another information to the network, we want to be able to change the code at as little places as possible. Up to discussion (tomorrow?)

Reply:
If we could do both with some smart decorator, it could be cool. Does `numpy.typing` allow shape hints like tensorflow?